src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1659,51 +1659,87 @@ protected override void InvokeDefaultAction(string path)
                             return;
                         }
 
-                        bool attributeFilter = true;
-                        bool switchAttributeFilter = true;
-                        bool filterHidden = false;           // "Hidden" is specified somewhere in the expression
-                        bool switchFilterHidden = false;     // "Hidden" is specified somewhere in the parameters
-
-                        if (null != evaluator)
+                        // Internal test code, run only if one of the
+                        // 'GciEnumerationAction' test hooks are set.
+                        if (InternalTestHooks.GciEnumerationActionDelete)
```

Comment:
I think we get already the list in `foreach (IEnumerable<FileSystemInfo> childList in target)` - so after that we can safely delete/remove test file.